cmd/minishift/cmd/root.go

```diff
@@ -251,3 +306,43 @@ func performPostUpdateExecution(markerPath string) error {
 
 	return nil
 }
+
+func ensureAllInstanceConfigPath(ConfigPath string) {
+	configDir := filepath.Dir(ConfigPath)
+	if err := os.MkdirAll(configDir, 0777); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", configDir))
+	}
+}
+
+func createDirForMinishiftRun(minishiftdirs dirs) {
+	dirPaths := reflect.ValueOf(minishiftdirs)
+
+	for i := 0; i < dirPaths.NumField(); i++ {
+		path := dirPaths.Field(i).Interface().(string)
+		if err := os.MkdirAll(path, 0777); err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", path))
+		}
+	}
+}
+
+//If there is no active profiles we need to set
+// minishift as the default profile if minishift profile exists
+func setDefaultActiveProfile() {
+	var setDefaultProfile bool
+	activeProfile := profileActions.GetActiveProfile()
+	if activeProfile == "" {
+		profileList := profileActions.GetProfileNameList()
+		for _, profile := range profileList {
+			if profile == constants.DefaultProfileName {
```

Comment:
why do we need to check for the active profile... just always set the profile?

Reply:
@gbraad we need to as there a chance user can run a command with --profile and it should supersede active profile.

Reply:
That logic should be more expli. Unable to extract that intention here.